delfin/drivers/dell_emc/vmax/client.py

```diff
@@ -287,6 +287,109 @@ def list_volumes(self, storage_id):
             LOG.error(msg)
             raise exception.StorageBackendException(msg)
 
+    def list_controllers(self, storage_id):
+        try:
+            # Get list of Directors
+            directors = self.rest.get_director_list(self.array_id,
+                                                    self.uni_version)
+            controller_list = []
+            for director in directors:
+                director_info = self.rest.get_director(
+                    self.array_id, self.uni_version, director)
+
+                status = constants.ControllerStatus.NORMAL
+                if director_info.get('availability', '').upper() != 'ONLINE':
+                    status = constants.ControllerStatus.OFFLINE
+
+                controller = {
+                    'name': 'director_'
```

Comment:
I think we dont need to append 'director'.. lets pick director_id as name, if there is no name available

Reply:
Thanks, Done.